### PR TITLE
chore: prepare `libp2p` `v0.51.2`

### PR DESCRIPTION
**Pull request by mxinden**
## Description

Depends-On: #3693.

## Notes & open questions

Unblocks @melekes. See https://github.com/libp2p/rust-libp2p/pull/3652#issuecomment-1485414961

Targets `master`. I will fast forward `v0.51` branch once this pull request is merged and release from `v0.51`.

I would like to include https://github.com/libp2p/rust-libp2p/pull/3693 first.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by melekes — APPROVED** [review]
👍

File: swarm/CHANGELOG.md [hunk 1]
```diff
@@ -12,12 +12,12 @@
 - Rename `NetworkBehaviourAction` to `ToSwarm`.
   A deprecated type-alias is provided to ease the transition.
   The new name is meant to better indicate the message-passing relationship between `Swarm` and `NetworkBehaviour`.
-  See [PR XXXX].
+  See [PR 3658].
```
Comment (thomaseizinger) [inline swarm/CHANGELOG.md:15]: Whoops, sorry about that!

**Review by thomaseizinger — APPROVED** [review]

File: swarm/CHANGELOG.md (hunk 1 above)
Comment (mxinden) [reply swarm/CHANGELOG.md:15]: Not a problem. Could have easily occurred to me too.

**Comment by mxinden** [comment]
Published and tagged the following crates:

- autonat-example-v0.1.0
- libp2p-allow-block-list-v0.1.0
- libp2p-connection-limits-v0.1.0
- libp2p-gossipsub-v0.44.2
- libp2p-mdns-v0.43.1
- libp2p-perf-v0.1.0
- libp2p-swarm-v0.42.1
- libp2p-v0.51.2
- libp2p-webrtc-v0.4.0-alpha.4
- relay-server-example-v0.1.0

**Comment by mxinden** [comment]
@thomaseizinger meta level, as before, super easy to cut a round of patch releases now.

**Comment by thomaseizinger** [comment]
> * autonat-example-v0.1.0
> * relay-server-example-v0.1.0

Why did we publish these two examples? I always thought of these as crates that we would just have in our workspace but not actually push to crates.io.

**Comment by mxinden** [comment]
Ah, sorry, confusing comment.

The examples are not published, only tagged.

https://crates.io/crates/autonat-example gives a 404.

`cargo release` tags them despite `publish = false`. I didn't bother excluding them from the `cargo release tag`. Do you feel strongly about excluding them?

**Comment by thomaseizinger** [comment]
Not at all, I just don't want them published 😁

**Pull request merged**